float32 sidelength
int32 angles
---
